Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/RecordNavPage.xaml.cs

```diff
@@ -40,197 +31,324 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
     /// </summary>
     public partial class RecordNavPage : Page
     {
-        bool isRecording = false;
+        public bool IsRecording = false;
         IWindowExplorer mWindowExplorerDriver;
-        private Activity mActParentActivity = null;
+        ePlatformType mActivityPlatform;
         Context mContext;
+        RecordingManager mRecordingMngr;
+        SingleItemTreeViewSelectionPage mApplicationPOMSelectionPage = null;
+        Agent mAgent;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="context"></param>
         public RecordNavPage(Context context)
         {
             InitializeComponent();
             mContext = context;
-            xWinGridUC.mContext = mContext;
+            mAgent = null;
+            context.PropertyChanged += Context_PropertyChanged;
+            InitMethods();
+        }
+
+        private void Agent_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Agent.Status))
+            {
+                SetControlsVisibility();
+            }
+        }
+
+        private void InitMethods()
+        {
+            string targetApp = string.Empty;
+            if (mContext.BusinessFlow.CurrentActivity == null || string.IsNullOrEmpty(mContext.BusinessFlow.CurrentActivity.TargetApplication))
+            {
```

Comment:
here is the code which you taking first Agent evendue the CurrentActivity is null, we should not take anything in case CurrentActivity is null, recording should be disabeled

Reply:
Done